jerry-debugger/src/lib/protocol-handler.ts

```diff
@@ -0,0 +1,251 @@
+      line: 1,
+      column: 1,
+      name: '',
+      source: '',
+      lines: [],
+      offsets: [],
+    }];
+  }
+
+  // for the temporary mollification of lint
+  unused() {
+    this.maxMessageSize;
+  }
+
+  decodeMessage(format: string, message: Uint8Array, offset: number) {
+    return decodeMessage(this.byteConfig, format, message, offset);
+  }
+
+  onConfiguration(data: Uint8Array) {
+    console.log('[Configuration]');
+    if (data.length !== 5) {
+      this.abort('configuration message wrong size');
+      return;
+    }
+
+    this.maxMessageSize = data[1];
+    this.byteConfig.cpointerSize = data[2];
+    this.byteConfig.littleEndian = Boolean(data[3]);
+    this.version = data[4];
+
+    if (this.byteConfig.cpointerSize !== 2 && this.byteConfig.cpointerSize !== 4) {
+      this.abort('compressed pointer must be 2 or 4 bytes long');
+    }
+
+    if (this.version !== JERRY_DEBUGGER_VERSION) {
+      this.abort('incorrect target debugger version detected: ' + this.version
+                 + ' expected: ' + JERRY_DEBUGGER_VERSION);
+    }
+  }
+
+  onByteCodeCP(data: Uint8Array) {
+    console.log('[Byte Code CP]');
+    const frame = this.stack.pop();
+    if (!frame) {
+      throw new Error('missing parser stack frame');
+    }
+
+    const byteCodeCP = this.decodeMessage('C', data, 1)[0];
+    const func = new ParsedFunction(byteCodeCP, frame);
+
+    this.functions[byteCodeCP] = func;
+    if (this.stack.length > 0) {
+      return;
+    }
+
+    // FIXME: it seems like this is probably unnecessarily keeping the
+    //   whole file's source to this point?
+    func.source = this.source.split(/\r\n[\r\n]/);
+    func.sourceName = this.sourceName;
+  }
+
+  onParseFunction(data: Uint8Array) {
+    console.log('[Parse Function]');
+    const position = this.decodeMessage('II', data, 1);
+    this.stack.push({
+      isFunc: true,
+      line: position[0],
+      column: position[1],
+      name: cesu8ToString(this.functionName),
+      source: this.source,
+      sourceName: this.sourceName,
+      lines: [],
+      offsets: [],
+    });
+    this.functionName = undefined;
+    return;
+  }
+
+  onBreakpointList(data: Uint8Array) {
+    console.log('[Breakpoint List]');
+    if (data.byteLength % 4 !== 1 || data.byteLength < 1 + 4) {
+      throw new Error('unexpected breakpoint list message length');
+    }
+
+    let array: Array<number>;
+    if (data[0] === SP.JERRY_DEBUGGER_BREAKPOINT_LIST) {
+      array = this.stack[this.stack.length - 1].lines;
+    } else {
+      array = this.stack[this.stack.length - 1].offsets;
+    }
+
+    for (let i = 1; i < data.byteLength; i += 4) {
+      array.push(this.decodeMessage('I', data, i)[0]);
+    }
+    return;
+  }
+
+  onSourceCode(data: Uint8Array) {
+    console.log('[Source Code]');
+    this.sourceData = concatUint8Arrays(this.sourceData, data);
+    if (data[0] === SP.JERRY_DEBUGGER_SOURCE_CODE_END) {
+      this.source = cesu8ToString(this.sourceData);
```

Comment:
Same here, do you still need `sourceData` after this point? If not, clean up and set to `undefined`?